util/src/main/java/com/google/cloud/hadoop/util/HttpTransportFactory.java

```diff
@@ -165,4 +174,68 @@ static URI parseProxyAddress(@Nullable String proxyAddress) {
           String.format("Invalid proxy address '%s'.", proxyAddress), e);
     }
   }
+
+  /** Wrapper class to have socketKeepAlive property while creating the socket */
+  @VisibleForTesting
+  static class SslKeepAliveSocketFactory extends SSLSocketFactory {
```

Comment:
can we make the class private?

Reply:
How will we test this class if I make this private?

HttpTransport doesn't have properties to easily verify socket keep alive is set or not.

Reply:
ack